### ISSUE TITLE
Improve upon model.train() return type consistency

### ISSUE THREAD

**Issue by SherlockMones** [frankkramer-lab/aucmedi#119]
Consolidate return types for transfer_learning and non-transfer_learning approaches:

In case for a tf-approach, we need to add the final results to the previous results as such

````
{
  'categorical_accuracy': [ ... ],
  'loss': [ ... ],
  'tf_categorical_accuracy': [ ... ],
  'tf_loss': [ ... ]
}
